ptp/ptp4u/server/server.go

```diff
@@ -106,6 +108,31 @@ func (s *Server) Start() error {
 		}
 	}()
 
+	// Drain check
+	go func() {
+		defer wg.Done()
+		for {
+			<-time.After(s.Config.DrainInterval)
+
+			var drain bool
+			for _, check := range s.Checks {
+				if check.Check() {
+					drain = true
+					break
+				}
+			}
+
+			if drain {
+				s.Drain()
+				s.Stats.SetDrain(1)
+				log.Warningf("killswitch engaged shifting traffic")
```

Comment:
I would move this to line 121 and specify which HC failed. Maybe removing break there so we always execute the checks and print the failed status of each.

Reply:
> specify which HC failed

These are not really health checks, it is not aimed to check the health of the system.

> and print the failed status of each

There is no failed status here, just a killswitch.

Reply:
> Maybe removing break there so we always execute the checks

Why would we want to execute all of them?

Reply:
Let me rephrase it - this section has nothing to do with killswitch anymore. This is now "Drain (ok not health) check failed". So "killswitch engaged shifting traffic" is simply not true here anymore.